File: Actividad4.c
/*
Nombre: Samara Mardueño Bernardino
Codigo:222727303
Descripcion: Este programa pide datos al usuario para medir su imc y dar un diagnostico en base a estos resultados en el programa se valida la edad el nombre peso altura
    
*/

#include<stdio.h>
#include<stdlib.h>
#include <string.h>

int main(){
    char nombre[30];
    int edad,i,esvalida=1;
    float kilos,altura,imc;
    
    printf("Cual es tu nombre Completo: ");
    fgets(nombre,sizeof(nombre),stdin);
    
    //Elimina el salto de linea
    for (i = 0; nombre[i] != '\0'; i++) {
        if (nombre[i] == '\n') {
            nombre[i] = '\0';
            break;
        }
    }
    //Valida que sea una cadena con letras y espacios
     for (i = 0; nombre[i] != '\0'; i++) {
        char c = nombre[i];
        if (!((c >= 'A' && c <= 'Z') || (c >= 'a' && c <= 'z')||(c == ' '))) {
            esvalida = 0; // Encontró algo que no es letra
            break;
        }
     }
     if(esvalida == 0){
                 printf("Tu nombre no es valido\n");
     }else{
                 printf("Tu nombre es valido\n");
                       }

     printf("Cual es tu edad: ");
     scanf("%d",&edad);
     
     if(edad > 1 && edad < 120){
          printf("Tu edad Es valida\n");
     }else{
          printf("Tu edad es invalida\n");     
     }
    
     printf("Cual es tu peso: ");
     scanf("%f",&kilos);
     
     if(kilos > 20 && kilos < 300){
         printf("Tu peso es valido\n");
     }else{
         printf("Tu peso es invalido\n");
     }
    
     printf("Cual es tu estatura en cm: ");
     scanf("%f",&altura);
    
     if((altura == (int)altura) && (altura > 50 && altura < 250)){
          printf("Tu estatura es valida\n");          
     }else{
           printf("Tu estatura no es valida\n");      
     }
     
     imc = kilos / ((altura/100) * (altura/100));
     
     
     system("pause"); 
     system("cls"); 
    
    printf("==========================================\n");
    printf("     INFORME DE DIAGNOSTICO IMC\n");
    printf("==========================================\n\n");
    printf("Paciente: %s\n",nombre);
    printf("Edad: %d anios\n",edad);
    printf("Peso: %.2f kg\n",kilos);
    printf("Altura %.2f m\n\n",(altura / 100));
    
    
    printf("RESULTADOS:\n\n");
    printf("IMC calculado: %.2f\n",imc);

    if (imc < 18.6) {
        printf("Diagnóstico: Tu IMC de %.2f se encuentra en el nivel de bajo peso\n", imc);
    } else if (18.4<imc < 25) {
        printf("Diagnóstico: Tu IMC de %.2f se encuentra en el peso normal\n", imc);
    } else if (24<imc < 30) {
        printf("Diagnóstico: Tu IMC de %.2f se encuentra en el nivel de sobrepeso\n", imc);
    } else if (29<imc < 35) {
        printf("Diagnóstico: Tu IMC de %.2f se encuentra en el nivel de obesidad tipo 1\n", imc);
    } else if (34<imc < 40) {
        printf("Diagnóstico: Tu IMC de %.2f se encuentra en el nivel de obesidad tipo 2\n", imc);
    } else {
        printf("Diagnostico: Tu IMC de %.2f se encuentra en el nivel de obesidad tipo 3\n", imc);
    }        
    printf("==========================================");
  
    
    getch();
    return 0;
    }

File: Actividad4.py
#Nombre: Samara Mardueño Bernardino
#Codigo:222727303
#Descripcion: Este programa pide datos al usuario para medir su imc y dar un diagnostico en base a estos resultados en el programa se valida la edad el nombre peso altura


nombre = input("¿Cuál es tu nombre completo?: ").strip()

# Validar que solo tenga letras y espacios
esvalida = all(c.isalpha() or c.isspace() for c in nombre)

if not esvalida:
    print("Tu nombre no es válido")
else:
    print("Tu nombre es válido")

# Edad
edad = int(input("¿Cuál es tu edad?: "))
if edad > 1 and edad < 120:
    print("Tu edad es válida")
else:
    print("Tu edad es inválida")

# Peso
kilos = float(input("¿Cuál es tu peso en kg?: "))
if kilos > 20 and kilos < 300:
    print("Tu peso es válido")
else:
    print("Tu peso es inválido")

# Altura
altura = float(input("¿Cuál es tu estatura en cm?: "))
if altura.is_integer() and altura > 50 and altura < 250:
    print("Tu estatura es válida")
else:
    print("Tu estatura no es válida")

# Calcular IMC (convertir altura a metros)
imc = kilos / ((altura / 100) ** 2)

# Informe
print("\n==========================================")
print("     INFORME DE DIAGNÓSTICO IMC")
print("==========================================\n")
print(f"Paciente: {nombre}")
print(f"Edad: {edad} años")
print(f"Peso: {kilos:.2f} kg")
print(f"Altura: {altura/100:.2f} m\n")
print("RESULTADOS:\n")
print(f"IMC calculado: {imc:.2f}")

# Diagnóstico según IMC
if imc < 18.6:
    print(f"Diagnóstico: Tu IMC de {imc:.2f} se encuentra en el nivel de bajo peso")
elif imc < 25:
    print(f"Diagnóstico: Tu IMC de {imc:.2f} se encuentra en el peso normal")
elif imc < 30:
    print(f"Diagnóstico: Tu IMC de {imc:.2f} se encuentra en el nivel de sobrepeso")
elif imc < 35:
    print(f"Diagnóstico: Tu IMC de {imc:.2f} se encuentra en el nivel de obesidad tipo 1")
elif imc < 40:
    print(f"Diagnóstico: Tu IMC de {imc:.2f} se encuentra en el nivel de obesidad tipo 2")
else:
    print(f"Diagnóstico: Tu IMC de {imc:.2f} se encuentra en el nivel de obesidad tipo 3")

print("==========================================")

